Makefile

```diff
@@ -48,22 +65,26 @@ linux-plugins:
     	--FIREHOSE_PLUGIN_BRANCH=${FIREHOSE_PLUGIN_BRANCH} \
     	--CLOUDWATCH_PLUGIN_CLONE_URL=${CLOUDWATCH_PLUGIN_CLONE_URL} \
     	--CLOUDWATCH_PLUGIN_TAG=${CLOUDWATCH_PLUGIN_TAG} \
-    	--CLOUDWATCH_PLUGIN_BRANCH=${CLOUDWATCH_PLUGIN_BRANCH}
-
-.PHONY: debug-base
-debug-base: linux-plugins
-	docker system prune -f
-	docker build --no-cache -t amazon/aws-for-fluent-bit:debug-base -f Dockerfile.debug-base .
-
-.PHONY: core
-core: debug-base
-	docker system prune -f
-	docker build -t amazon/aws-for-fluent-bit:core -f Dockerfile.core .
-
-.PHONY: valgrind
-valgrind: debug-base
-	docker system prune -f
-	docker build -t amazon/aws-for-fluent-bit:valgrind -f Dockerfile.core .
+    	--CLOUDWATCH_PLUGIN_BRANCH=${CLOUDWATCH_PLUGIN_BRANCH} \
+		--DOCKER_BUILD_FLAGS=${DOCKER_BUILD_FLAGS}
+
+.PHONY: main-debug
+main-debug: main-debug-base
+	docker build $(DOCKER_BUILD_FLAGS) -t amazon/aws-for-fluent-bit:debug-fs       -f ./scripts/dockerfiles/Dockerfile.main-debug-fs .
+	docker build $(DOCKER_BUILD_FLAGS) -t amazon/aws-for-fluent-bit:debug-s3       -f ./scripts/dockerfiles/Dockerfile.main-debug-s3 .
+	docker build $(DOCKER_BUILD_FLAGS) -t amazon/aws-for-fluent-bit:debug-valgrind -f ./scripts/dockerfiles/Dockerfile.main-debug-valgrind .
```

Comment:
I think this is fine, but I was kind of expecting separate targets to build each type of debug image. Say you just want valgrind for example.

Reply:
They build super quickly, so I though why not just put all of them together. The part that takes a long time to build is the main-debug-base

Reply:
yea its fine